client/blocks/stats-navigation/index.js

```diff
@@ -123,36 +130,29 @@ const SelectNav = ( {
 	);
 };
 
-const TabNav = ( { validNavItems, interval, slugPath, adminUrl, selectedItem, showLock } ) => {
-	const tabs = validNavItems.map( ( item ) => {
-		const navItem = navItems[ item ];
-		const intervalPath = navItem.showIntervals ? `/${ interval || 'day' }` : '';
-		const itemPath = `${ navItem.path }${ intervalPath }${ slugPath }`;
-		return {
-			name: item,
-			title: navItem.label + ( navItem.paywall && showLock ? ' 🔒' : '' ),
-			className: 'stats-navigation__' + item,
-			path: itemPath,
-		};
-	} );
-
+/**
+ * @param { { tabs: StatsNavItem[], selectedItemName: keyof typeof allNavItems } } props
+ */
+const TabNav = ( { tabs, selectedItemName } ) => {
 	return (
 		<TabPanel
 			className="stats-navigation__tabs"
 			tabs={ tabs }
-			onSelect={ ( tabName ) => {
+			onSelect={ ( newSelectedTabName ) => {
```

Comment:
```suggestion
			onSelect={ ( selectedTabName ) => {
```
Is the naming `selectedTabName` enough for the meaning?

Reply:
Yep!

But, we have a prop in the `TabNav` called `selectedItemName` (which I think should be renamed to `selectedTabName` for consistency). That's why added this `new` prefix.

Reply:
I feel like `selectedItemName` is still good enough as it comes from the selected nav item, which can be compared with `selectedTabName`.
```
if ( selectedTabName === selectedItemName ) {
```